instrumentation/apache-httpclient/apache-httpclient-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v4_0/ApacheHttpClientNetAttributesExtractor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachehttpclient.v4_0;
+
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.net.URI;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class ApacheHttpClientNetAttributesExtractor
+    extends NetAttributesExtractor<HttpUriRequest, HttpResponse> {
+
+  @Override
+  protected String transport(HttpUriRequest request) {
+    return SemanticAttributes.NetTransportValues.IP_TCP;
+  }
+
+  @Override
+  protected @Nullable String peerName(HttpUriRequest request, @Nullable HttpResponse response) {
+    return request.getURI().getHost();
+  }
+
+  @Override
+  protected Long peerPort(HttpUriRequest request, @Nullable HttpResponse response) {
+    URI uri = request.getURI();
+    int port = uri.getPort();
+    if (port != -1) {
+      return (long) port;
+    }
+    switch (uri.getScheme()) {
+      case "http":
+        return 80L;
+      case "https":
+        return 443L;
+    }
+    return null;
```

Comment:
do you think I should extract `UriNetAttributesExtractor` similar to `InetSocketAddressNetAttributesExtractor`?

Reply:
Is it correct to say URI will be more common in automatic instrumentation vs library instrumentation since it's a result of not having access to the internals? If so, possibly not, to encourage library authors adding OTel support to expose the more useful type.

Reply:
I'm not sure how common yet, will know soon 😁 (won't extract anything common for now and we can revisit later)

Reply:
There's one advantage to using URI in the instrumenter API: if we used it everywhere then we could have encapsulate the userInfo sanitization logic in the attribute extractor (well, we still can do it right now, but with a `String url()` method we'd have to have one more type conversion)